modules/engines/engine-core/src/main/java/com/sap/xsk/modificators/XSKProjectFilesModificator.java

```diff
@@ -11,53 +11,108 @@
-    String fileName = projectFile.getName();
-    String fileExtension = StringUtils.substringAfterLast(fileName, ".");
+    String fileExtension = getProjectFileExtension(projectFile);
 
     switch (fileExtension) {
-      case "calculationview":
-      case "hdbcalculationview": {
-        modifyCalculationViewFile(projectFile);
+      case "xsjs":
+      case "xsjslib":
+      case "hdbprocedure":
+      case "hdbtablefunction": {
+        replaceSessionUser(projectFile);
         break;
       }
       case "analyticprivilege":
       case "hdbanalyticprivilege": {
+        replaceSessionUser(projectFile);
         modifyAnalyticPrivilegeFile(projectFile);
         break;
       }
     }
   }
 
-  private void modifyCalculationViewFile(IFile calculationViewFile) {
-    byte[] currentContent = calculationViewFile.getContent();
-    // modify
-    calculationViewFile.setContent(currentContent);
+  private void addNewProjectFileIfNecessary(IProject project) {
+  }
+
+  private void replaceSessionUser(IFile projectFile) {
+    byte[] currentContent = projectFile.getContent();
+    projectFile.setContent(new String(currentContent).replace("SESSION_USER", "SESSION_CONTEXT(APPLICATIONUSER)").getBytes());
   }
 
   private void modifyAnalyticPrivilegeFile(IFile analyticPrivilegeFile) {
     byte[] currentContent = analyticPrivilegeFile.getContent();
-    analyticPrivilegeFile.setContent(currentContent);
+
+    if (currentContent.length != 0) {
+      try {
+        String xslt = IOUtils.toString(
+            XSKProjectFilesModificator.class.getClassLoader().getResourceAsStream(XSLT_RESOURCE_PATH),
+            StandardCharsets.UTF_8);
+
+        StringWriter writer = new StringWriter();
+        StreamResult result = new StreamResult(writer);
+
+        TransformerFactory factory = TransformerFactory.newInstance();
+        Source xsltSource = new StreamSource(new StringReader(xslt));
+        Transformer transformer = factory.newTransformer(xsltSource);
+
+        Source contentSource = new StreamSource(new StringReader(new String(currentContent)));
+        transformer.transform(contentSource, result);
+
+        String processedAnalyticPrivilege = writer.toString();
+
+        analyticPrivilegeFile.setContent(processedAnalyticPrivilege.getBytes());
+      } catch (Exception e) {
+        logger.error("Incorrect analytic privilege file content format.");
+      }
+
+    }
   }
 
-  private void addNewProjectFileIfNecessary(IProject project) {
-    // // public IFile createFile(String path, byte[] content) {}
-    // // public IFile createFile(String path, byte[] content, boolean isBinary, String contentType) {}
-    //
-    // project.createFile()
+  private String getProjectFileExtension(IFile projectFile) {
+    return StringUtils.substringAfterLast(projectFile.getName(), ".");
+  }
+
+  public static String processModelUri(String value) {
+    int start = 0;
+
+    if (value.indexOf("/") != -1) {
```

Comment:
for better readability we can extract "value.indexOf("/") == -1" in a local variable named "uriIsNotWithPackageAndFolder" and add one If statement if(uriIsNotWithPackageAndFolder) { return value;} and leave existing logic of processing out of the if.